src/main/java/edu/kit/kastel/sdq/artemis4j/grading/Assessment.java

```diff
@@ -0,0 +1,427 @@
+     */
+    public List<Annotation> getAnnotations(RatingGroup ratingGroup) {
+        return this.annotations.stream().filter(a -> a.getMistakeType().getRatingGroup().equals(ratingGroup)).toList();
+    }
+
+    /**
+     * Adds a non-custom manual annotation to the assessment.
+     * @param mistakeType Must not be a custom mistake type
+     * @param filePath
+     * @param startLine
+     * @param endLine
+     * @param customMessage May be null if no custom message is provided
+     */
+    public void addPredefinedAnnotation(MistakeType mistakeType, String filePath, int startLine, int endLine, String customMessage) {
+        if (mistakeType.isCustomAnnotation()) {
+            throw new IllegalArgumentException("Mistake type is a custom annotation");
+        }
+
+        var source = this.correctionRound == 0 ? Annotation.AnnotationSource.MANUAL_FIRST_ROUND : Annotation.AnnotationSource.MANUAL_SECOND_ROUND;
+        this.annotations.add(new Annotation(mistakeType, filePath, startLine, endLine, customMessage, null, source));
+    }
+
+    /**
+     * Adds a custom manual annotation to the assessment.
+     * @param mistakeType Must be a custom mistake type
+     * @param filePath
+     * @param startLine
+     * @param endLine
+     * @param customMessage May not be null
+     * @param customScore
+     */
+    public void addCustomAnnotation(MistakeType mistakeType, String filePath, int startLine, int endLine, String customMessage, double customScore) {
+        if (!mistakeType.isCustomAnnotation()) {
+            throw new IllegalArgumentException("Mistake type is not a custom annotation");
+        }
+
+        var source = this.correctionRound == 0 ? Annotation.AnnotationSource.MANUAL_FIRST_ROUND : Annotation.AnnotationSource.MANUAL_SECOND_ROUND;
+        this.annotations.add(new Annotation(mistakeType, filePath, startLine, endLine, customMessage, customScore, source));
+    }
+
+    public void addAutograderAnnotation(MistakeType mistakeType, String filePath, int startLine, int endLine, String explanation) {
+        var customScore = mistakeType.isCustomAnnotation() ? 0.0 : null;
+        this.annotations.add(new Annotation(mistakeType, filePath, startLine, endLine, explanation, customScore, Annotation.AnnotationSource.AUTOGRADER));
+    }
+
+    /**
+     * Removes an annotation from the assessment. If the annotation is not present, nothing happens.
+     *
+     * @param annotation
+     */
+    public void removeAnnotation(Annotation annotation) {
+        this.annotations.remove(annotation);
+    }
+
+    /**
+     * Clears all annotations from the assessment.
+     */
+    public void clearAnnotations() {
+        this.annotations.clear();
+    }
+
+    private void internalSaveOrSubmit(boolean shouldSubmit, Locale locale) throws AnnotationMappingException, ArtemisNetworkException {
+        log.info("Packing assessment for artemis");
+        var feedbacks = this.packAssessmentForArtemis(locale);
```

Comment:
The Packing of assessments should be independent from the locale. The multi-language support is just an UI thing. We can change this in the future, but for now it should be fixed somehow. Basically this ensures that even a missing translation in the config does not break anything.

Reply:
We had a lengthy discussion about that on Element. Nothing will break in case of a missing translation. Some students may prefer their feedback in English, and since both button texts and Autograder annotations are already available in English, we could provide English feedback. However, we currently have no reliable way to detect the student's preference, so the default will be German.

If you prefer so, I can of course remove the functionality for now.